### PR TITLE
Asynchronous Dials To Peers 

### PR DESCRIPTION
**Pull request by nisdas**
This PR changes how we connect to peers, spinning each dial into a separate go-routine to make them non blocking. Also removes the exclusion list we were using an instead uses our peer handler, to handle bad peers.

### CODE REVIEW

File: beacon-chain/p2p/discovery_test.go [hunk 1]
```diff
@@ -161,6 +161,7 @@ func TestStaticPeering_PeersAreAdded(t *testing.T) {
 	s.Start()
 	s.dv5Listener = &mockListener{}
 	defer s.Stop()
+	time.Sleep(100 * time.Millisecond)
```
Comment (terencechain) [inline beacon-chain/p2p/discovery_test.go]: Instead of sleep here, would `sync.WaitGroup` be better?

Comment (nisdas) [reply beacon-chain/p2p/discovery_test.go]: wait group would only work if we could use it in each go-routine, however the method is enclosed in `s.Start()` so it isnt accesible to `sync.WaitGroup` from the test

Comment (terencechain) [reply beacon-chain/p2p/discovery_test.go]: Hm I see. So no way to avoid the sleep then?

Comment (nisdas) [reply beacon-chain/p2p/discovery_test.go]: Unfortunately, not. Especially for the current test.

**Pull request merged**